src/myrtlespeech/builders/task_config.py

```diff
@@ -110,18 +112,32 @@ def target_transform(target):
     )
 
     shuffle = task_config.train_config.shuffle_batches_before_every_epoch
-    train_loader = torch.utils.data.DataLoader(
-        dataset=train_dataset,
-        batch_sampler=RandomBatchSampler(
-            indices=range(len(train_dataset)),
-            batch_size=task_config.train_config.batch_size,
-            shuffle=shuffle,
-            drop_last=False,
-        ),
-        num_workers=num_workers,
-        collate_fn=seq_to_seq_collate_fn,
-        pin_memory=torch.cuda.is_available(),
-    )
+    if task_config.train_config.sortagrad:
+        train_loader = torch.utils.data.DataLoader(
+            dataset=train_dataset,
+            batch_sampler=SortaGrad(
+                indices=range(len(train_dataset)),
+                batch_size=task_config.train_config.batch_size,
+                shuffle=shuffle,
+                drop_last=False,
+            ),
```

Comment:
I did the exact same thing you have written, but unfortunately black was complaining during the pre-commit because he thinks the variable batch_sampler should be always a SortaGrad or a SequentialRandomSampler (he thinks there is a type inconsistency). I couldn't find any solution to this problem other than duplicate the code in the if-else statements.

Reply:
I think you can define the types in the main body in this case:
```
from typing import Union
batch_sampler: Union[SortaGrad, SequentialRandomSampler]
if task_config.train_config.sortagrad:
    batch_sampler = SortaGrad(...)
    collate_fn = seq_to_seq_collate_fn_sorted
else:
    batch_sampler = SequentialRandomSampler(...)
    collate_fn =  seq_to_seq_collate_fn
```

Reply:
It seems it is working, but I couldn't avoid also the repetition of the batch sampler initialization